src/main/java/uk/ac/cam/cl/dtg/isaac/api/services/AssignmentService.java

```diff
@@ -39,24 +41,25 @@ public AssignmentService(final UserAccountManager userManager) {
         this.userManager = userManager;
     }
 
-    public <T extends IAssignmentLike> void augmentAssignerSummaries(Collection<T> assignments) throws SegueDatabaseException {
+    public <T extends IAssignmentLike> void augmentAssignerSummaries(final Collection<T> assignments) throws SegueDatabaseException {
         Map<Long, UserSummaryDTO> userSummaryCache = new HashMap<>();
 
+        // Iterating over the owner IDs allows us to cache "no user found" errors without querying database each time:
+        List<Long> ownerUserIds = assignments.stream().map(IAssignmentLike::getOwnerUserId).distinct().collect(Collectors.toList());
+        for (Long ownerUserId: ownerUserIds) {
+            try {
```

Comment:
This doesn't try to get the summary from the cache anymore. Is this correct?

Reply:
Erm, yeah, because of the `.distinct()` each user will only be looked up once, and this is how we populate the cache to use later when we loop by assignment (where there are lots of duplicated owner users a chache is good for).

I think this diff is best viewed separately, if I am honest; the combined diff muddles it all up horribly!

Reply:
Yeah, that does make it a bit more readable indeed…